src/components/Login/LoginOTPCard.tsx

```diff
@@ -69,11 +75,43 @@ export const LoginOTPCard: FunctionComponent<LoginOTPCard> = ({
     try {
       const response = await validateOTP(otp, mobileNumber, codeKey, endpoint);
       setIsLoading(false);
-      setAuthInfo(response.sessionToken, response.ttl.getTime(), endpoint);
-      navigation.navigate("CollectCustomerDetailsScreen");
+      const versionResponse = await getEnvVersion(
+        response.sessionToken,
+        endpoint
+      );
+
+      // Toggle between different environments
+      // using the DIST_ENV variable from features
+
+      switch (versionResponse.features.DIST_ENV) {
+        case "VOUCHER":
+        case "TOKEN":
+          setAuthInfo(response.sessionToken, response.ttl.getTime(), endpoint);
+          setFeatures(versionResponse.features);
+          setProducts(versionResponse.policies);
+          navigation.navigate("CollectCustomerDetailsScreen");
+          break;
+
+        // TODO: Integration with merchant flow
+        // case "MERCHANT":
```

Comment:
Are we going to have separate environments for each? 

Is there a way we can generalise the two types of flows for scalability?

Reply:
E.g. right now TT and CDC voucher shares the same flow. So why do we call it "Voucher"?

Reply:
putting the same comment here: hmm yeah i think being general defo more scalable, distributions in the future (FFTH) will end up having the same flow as cdc voucher and tt token, im okay with naming the environment smth general like `COLLECTION` or `DEFAULT`?